reinforcement_learning/rlclientlib/logger/event_batcher.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+#include "event_queue.h"
+#include "ranking_event.h"
+#include "utility/data_buffer.h"
+
+namespace reinforcement_learning {
+  namespace utility {
+    class event_batcher {
+    public:
+      event_batcher();
+      void batch_serialize(data_buffer& oss, size_t& remaining, event_queue<ranking_event>& queue, size_t _send_high_water_mark);
+      void batch_serialize(data_buffer& oss, size_t& remaining, event_queue<outcome_event>& queue, size_t _send_high_water_mark);
+
+      template<typename TEvent>
+      void batch_serialize(data_buffer& oss, size_t& remaining, event_queue<TEvent>& queue, size_t _send_high_water_mark);
+
+    private:
+      size_t _batch_size = 400;
```

Comment:
Can you double check what is going on between _batch_size and _send_high_water_mark? Looks like they have to be collapsed into one thing.

Reply:
This is known issue and had been discussed for a while. The problem here is we won't know the flatbuffer size before we call the build to serialize it.

It is possible to remove _send_high_water_mark but this could be a breaking change. I remembered Rajan mentioned we will have plan to do fix length transport. In this case, this may be still needed.

With Rajan's proposed preamble header, we may be able to solve it.

---
In reply to: [228611953](https://github.com/VowpalWabbit/vowpal_wabbit/pull/1653#discussion_r228611953) [](ancestors = 228611953)

Reply:
Oh I see, they are both used in a different way - then please see the new comment on event_batcher.cc